packages/apps/send-transactional-emails/src/functios-lib/handle-orders.ts

```diff
@@ -0,0 +1,254 @@
+                      order,
+                      lang,
+                      customMessage,
+                    );
+                  }
+                } else {
+                  if (!order.fulfillment_status) {
+                    order.fulfillment_status = {
+                      current: checkStatus as Exclude<Orders['fulfillment_status'], undefined>['current'],
+                    };
+                  }
+
+                  // TODO: appData[ ] is in SnakeCase
+                  templateId = appData[checkStatus] && appData[checkStatus].templateId;
+                  html = await transactionalMails[triggerStatus](
+                    store,
+                    customer,
+                    order,
+                    lang,
+                    customMessage,
+                  );
+                }
+
+                if (html) {
+                  // const template = transactionalMails.template(triggerStatus)
+                  // TODO: Need to change (PR) in transactionalMails?
+                  try {
+                    const to = [{
+                      name: customer.display_name,
+                      email: customer.main_email,
+                    }];
+                    await email.send(
+                      {
+                        to,
+                        subject,
+                        html,
+                        templateId,
+                        templateData: {
+                          store,
+                          customer,
+                          order,
+                          lang,
+                          customMessage,
+                        },
+                        // template
+                      },
+                    );
+                    // logger.log('>> Email sent');
+                    if (subresource === 'payments_history' || subresource === 'fulfillment') {
+                      await updateOrderSubresource(
+                        orderId,
+                        subresource,
+                        lastValidRecord,
+                        insertedId,
+                      );
+                    }
+                    return { message: 'SUCCESS' };
+                  } catch (err) {
+                    return handleErr(err);
+                  }
+                } else {
+                  return { message: 'SUCCESS' };
+                }
+              } else {
+                // logger.log('>> Customer already notified of this status')
+                return { message: 'SUCCESS' };
+              }
+            } else {
+              // TODO: check message
+              return { status: 404, message: 'Not Found Status' };
+            }
+          });
+        } else {
+          // logger.error('>> Not Found Customer')
+          return { status: 404, message: 'Not Found Customer' };
+        }
+      } else {
+        // logger.error('>> Not found Order or Store')
+        return { status: 404, message: 'Not found Order or Store' };
```

Comment:
Ixe 😄

Reply:
Só pra constar, esse `return { status: 404 ... }` também é inútil aqui mas nesse caso não queremos um error throw porque não é uma falha da função, é uma "falha do cliente" (em REST 4xx é falha do cliente, isso aqui também não geraria uma retentativa do webhook)

Reply:
Quer dizer, o payload do evento parece não ser válido, uma retentativa com ele não resolve o problema, só executaria várias (inifinitas ? 😬 ) vezes a função com o payload inválido

Reply:
Isso aqui vira apenas um `logger.warn`

Reply:
Tipo aí https://github.com/ecomplus/cloud-commerce/blob/main/packages/apps/tiny-erp/src/event-to-tiny.ts#L45

Reply:
E nesse caso o que resta é torcer pro "cliente" ver o erro no log e corrigir o problema pros próximos eventos funcionarem corretamente